src/imp/schannel.rs

```diff
@@ -91,14 +91,11 @@ impl Identity {
             }
         };
         
-        let mut store = CertStore::open_current_user("RustMy").unwrap();
+        let mut store = CertStore::open_current_user("RustMy")?;
 
         let existing_cert = CertStore::find_existing_cert_and_key(&mut identity, &mut store)?;
-        match existing_cert {
-            Some(x) => {
-                identity = store.add_cert(&x, CertAdd::ReplaceExistingInheritProperties)?;
-            }
-            None => ()
+        if let Some(x) = existing_cert {
+            identity = store.add_cert(&x, CertAdd::ReplaceExistingInheritProperties)?;
```

Comment:
This store needs to be the provider store. We are replacing with the existing cert's key provider info.